### ISSUE TITLE
[3.3] Hyperparameter Tuning [MANUAL RESEARCH]

### ISSUE THREAD

**Issue by poltergeistjoya** [poltergeistjoya/Vacant_Lot_Detection#19]
## Summary
Run independent hyperparameter sweep experiments. Parallelizable—3 hours each on T4. Total: ~12 runs.

### Experiment 1: pos_weight (class imbalance handling)
```
Values: [5, 10, 20, 50]
Fix: bce_weight=0.5, dice_weight=0.5, lr=1e-4
Rationale: Higher pos_weight upweights false negatives (missed vacant pixels)
Expected: Increasing beyond 10 may hurt precision or cause instability
```

### Experiment 2: Loss Ratio (BCE vs. Dice balance)
```
Values: [(0.3, 0.7), (0.5, 0.5), (0.7, 0.3)]
         (bce_weight, dice_weight)
Fix: pos_weight=10, lr=1e-4
Rationale: Dice-heavy favors region recall; BCE-heavy favors per-pixel precision
Expected: Dice-heavy (0.3, 0.7) likely best for imbalanced segmentation
```

### Experiment 3: Learning Rate
```
Values: [1e-3, 3e-4, 1e-4, 3e-5]
Fix: pos_weight=10, bce_weight=0.5, dice_weight=0.5
Rationale: Stability vs. convergence speed
Expected: 1e-4 or 3e-4 likely optimal; 1e-3 may overshoot
```

### Experiment 4: Architecture Comparison
```
Run 1: U-Net (ResNet34) with best hyperparams from 1–3
Run 2: DeepLabV3+ (ResNet34) with same hyperparams
Rationale: DeepLabV3+ multi-scale ASPP often improves semantic segmentation
Expected: DeepLabV3+ +2–5% IoU improvement
```

## Execution Plan

1. **Baseline (from 3.1):** U-Net, pos_weight=10, (0.5, 0.5), lr=1e-4
2. **Run 1–4:** pos_weight sweep (4 runs, ~3h each)
3. **Run 5–7:** Loss ratio sweep (3 runs)
4. **Run 8–11:** Learning rate sweep (4 runs)
5. **Run 12:** DeepLabV3+ with best hyperparams

**Parallel execution on Kaggle:** Sequential across sessions (checkpointing handles interruption). TensorBoard logs saved locally for analysis.

## Evaluation
- Primary metric: **Val IoU**
- Secondary: Dice, Precision, Recall, Kappa
- Record all runs in a spreadsheet:
  | Run | Architecture | pos_weight | (BCE, Dice) | LR | Val IoU | Best |
  |-----|-------------|-----------|-----------|----|---------| ----|

## Visualization & Analysis
1. **Line plot:** pos_weight vs. Val IoU (from Exp 1)
2. **Line plot:** Loss ratio vs. Val IoU (from Exp 2)
3. **Line plot:** Learning rate vs. Val IoU (from Exp 3)
4. **Bar chart:** Architecture comparison (U-Net vs. DeepLabV3+)
5. **Heatmap (optional):** 2D grid of (pos_weight, loss_ratio) with Val IoU

## Acceptance Criteria
- [ ] 12+ runs completed
- [ ] All runs logged with TensorBoard
- [ ] Spreadsheet of results created
- [ ] Best hyperparameters identified (e.g., pos_weight=20, (0.3, 0.7), lr=3e-4)
- [ ] Visualizations generated
- [ ] Written summary: why best config works

## Cost Analysis
| Platform | Cost/Run | 12 Runs |
|----------|----------|---------|
| Kaggle | $0 | **$0** (within 120h/month quota) |
| GCP Spot T4 | ~$0.33 | ~$4 |
| GCP On-demand T4 | ~$1.05 | ~$13 |

## Dependencies
- Issue #18 (3.2) — label noise strategy decided
- Issue #17 (3.1) — baseline established

## Manual Research Required
- TensorBoard analysis: which configs converge fastest?
- Cross-val analysis: which config generalizes best (low train-val gap)?
- Failure mode inspection: do different hyperparams shift error distribution?
